src/integrations/watchers/indexable-author-watcher.php

```diff
@@ -90,4 +92,25 @@ public function build_indexable( $user_id ) {
 			$indexable->save();
 		}
 	}
+
+	/**
+	 * Build a user indexable if the updated meta is in a list of watched meta keys.
+	 *
+	 * @param int    $meta_id   ID of updated metadata entry.
+	 * @param int    $object_id Post ID.
+	 * @param string $meta_key  Metadata key.
+	 *
+	 * @return void
+	 */
+	public function process_meta_change( $meta_id, $object_id, $meta_key ) {
+		$watched_meta_keys = [
+			'wpseo_title',
+			'wpseo_metadesc',
+			'wpseo_noindex_author',
+		];
+
+		if ( in_array( (string) $meta_key, $watched_meta_keys, true ) ) {
+			$this->build_indexable( $object_id );
```

Comment:
We should test if this doesn't trigger a double build when editing a user.

Reply:
It will trigger _up to_ 3 times, but only for the values you changed.
https://github.com/Yoast/wordpress-seo/blob/951a4b9626d8752085b2507a51a762130997f80b/admin/class-admin-user-profile.php#L72-L76
I can see a solution being scheduling an update when the meta changes and performing the update on another hook, like `personal_options_update` but I fear that we'd miss out on meta changes that are made outside of the profile edit page.
Alternatively, a shutdown hook would work, but that would not work for our integration tests 🙃 

Do you think the current approach is acceptable @herregroen?

Reply:
Discussed in a call wit Herre: we can introduce a scheduler that deals with scheduling tasks like rebuilding indexables and perform those actions on shutdown by default. In integrations tests we can simulate the shutdown by performing scheduled tasks manually. 
Right now the current implementation is good enough, but if similar issues arise, we could consider the scheduler.

Reply:
The above is tracked in `AR-288`